test/e2e/service.go

```diff
@@ -570,29 +575,10 @@ var _ = Describe("Services", func() {
 			Failf("Failed to reach load balancer at original ingress after updating its port: %+v", service)
 		}
 
-		By("changing service " + serviceName + " back to type=ClusterIP")
-		service, err = updateService(f.Client, f.Namespace.Name, serviceName, func(s *api.Service) {
-			s.Spec.Type = api.ServiceTypeClusterIP
-			s.Spec.Ports[0].NodePort = 0
-		})
-		Expect(err).NotTo(HaveOccurred())
-
-		if len(service.Status.LoadBalancer.Ingress) != 0 {
-			Failf("got unexpected len(Status.LoadBalancer.Ingress) for NodePort service: %v", service)
-		}
-		if service.Spec.Type != api.ServiceTypeClusterIP {
-			Failf("got unexpected Spec.Type for back-to-ClusterIP service: %v", service)
-		}
-		if len(service.Spec.Ports) != 1 {
-			Failf("got unexpected len(Spec.Ports) for back-to-ClusterIP service: %v", service)
-		}
-		port = service.Spec.Ports[0]
-		if port.NodePort != 0 {
-			Failf("got unexpected Spec.Ports[0].nodePort for back-to-ClusterIP service: %v", service)
-		}
+		removeExternalLoadBalancer(f, service)
 
 		// Wait for the load balancer to be destroyed asynchronously
-		service, err = waitForLoadBalancerDestroy(f.Client, serviceName, f.Namespace.Name)
+		service, err = waitForLoadBalancerDestroy(f.Client, serviceIP, strconv.Itoa(servicePort), serviceName, f.Namespace.Name)
```

Comment:
`removeExternalLoadBalancer` calls `waitForLoadBalancerDestroy`, so probably don't need this call anymore. (or the `servicePort`/`serviceIP` changes you have here.)